File: CODE/RetrieveBoxscoresTable.py
import sqlite3
import pandas as pd
import os

# Connect
path_to_database = os.path.join(os.getenv("EXAMPLES"), "Boxscores.db")

# Context manager handles connection.
with sqlite3.connect(path_to_database) as con:
    # read_sql function executes SQL query
    query_text = "SELECT * FROM Boxscores"
    print(f"Executing query: {query_text}")

    results_df = pd.read_sql(query_text, con)

    print(results_df)
    print("Done")

File: CODE/RetrieveBoxscoresTable.R
library(RSQLite)
library(DBI)
library(dplyr)

# Connect
path_to_database <- file.path(Sys.getenv("EXAMPLES"), "Boxscores.db")

con <- dbConnect(RSQLite::SQLite(), path_to_database)
  
# Get ALL data from Boxscores table
query <- tbl(con, "Boxscores") |>
  select(everything())
  
  # Look at SQL commands that this query builds
show_query(query)

# Actually get the Boxscores data -- using collect against the query
results_df <- query |>
  collect()
  
dbDisconnect(con)

print(results_df, width = Inf)
print("Done")